### ISSUE TITLE
Reorganize options shown in Preferences and menus

### ISSUE THREAD

**Issue by ccordoba12** [spyder-ide/ux-improvements#75]
Right now there are some entries in Preferences, e.g. the Editor one

![image](https://user-images.githubusercontent.com/365293/187800529-2287109d-6d32-4b89-8a09-eca324479e65.png)

![image](https://user-images.githubusercontent.com/365293/187800481-92c26b30-5c96-4e09-a3d6-c3b001816f49.png)

and menus, e.g. the Source one

![image](https://user-images.githubusercontent.com/365293/187800623-cd331c34-a6bd-4a4f-a67a-67c84e57b14e.png)

which have way too many options without a clear structure.

So I propose we should do the following:

1. Audit all Preferences' entries and menus to see how to present the information in them better.
2. Reorganize entries in Preferences so they are always shown in the same order.

**Comment by ccordoba12** [comment]
Another idea (but this is open to discussion) would be to add a question mark widget next to options in Preferences that have associated tooltips. That way, when users click on that widget, we'd show the info that we currently show in tooltips.

I think that would help them to better understand some complex options, which have good explanatory tooltips. However, those are probably missed because it's hard to know that you can get info on how an option works by hovering over it.

**Comment by CAM-Gerlach** [comment]
> Audit all Preferences' entries and menus to see how to present the information in them better.

My Qt-fu to actually implement the changes is not so stronk (though if its just moving section groupings and orderings around, it would be possible), but auditing what we have and proposing a specific, logical, user-friendly order (and clearer and more consistent UI text) is right within my wheelhouse, as I've been wanting to do that for a while.

> Reorganize entries in Preferences so they are always shown in the same order.

Here, do you mean the order of the preference panes in the left sidebar of the preferences? I believe there was a discussion about it before, but due to the changes in Spyder 5 with it being dependent on plugin load order (IIRC) the ordering doesn't really make clear sense, as it is neither alphabetical or directly following some type of progression (e.g. the same as in the `Panes` menu, with the Editor at the top). I suggest just following the order in the `Panes` menu, though perhaps that could also use a bit of re-ordering, which I could take a closer look at if desired.

> Another idea (but this is open to discussion) would be to add a question mark widget next to options in Preferences that have associated tooltips. That way, when users click on that widget, we'd show the info that we currently show in tooltips.

Very good points and that's a common and very helpful UI pattern, so long as it isn't complex to implement.

**Comment by ccordoba12** [comment]
> but auditing what we have and proposing a specific, logical, user-friendly order (and clearer and more consistent UI text) is right within my wheelhouse, as I've been wanting to do that for a while

Great! Please leave your suggestions here so we can take a look at them.

> Here, do you mean the order of the preference panes in the left sidebar of the preferences?

Correct.

>  I suggest just following the order in the Panes menu, though perhaps that could also use a bit of re-ordering, which I could take a closer look at if desired.

Good suggestion, I agree with it.

> Very good points and that's a common and very helpful UI pattern, so long as it isn't complex to implement.

I don't think it'll be hard to implement (that's why I suggested it).

**Comment by dalthviz** [comment]
Note: To show widgets help info the `WhatsThisMode` could be used: https://doc.qt.io/qt-5/qwhatsthis.html#enterWhatsThisMode

**Comment by ccordoba12** [comment]
What does that mode do @dalthviz?

**Comment by dalthviz** [comment]
When pressing the `?` (that usually is at the top of some dialogs) shows a message for the widgets/actions that have used [`setWhatsThis`](https://doc.qt.io/qt-5/qwidget.html#whatsThis-prop) to set some help text (which for now could be the tooltip text available). For example for the `buffer}  config of the IPython Console using the current tooltip text it has it would look something like:

![help](https://user-images.githubusercontent.com/16781833/234389620-f2d5b2ce-1729-4125-a62d-dc1e8eb34482.gif)

**Comment by ccordoba12** [comment]
Ok, that looks interesting but not easily discoverable.

@conradolandia showed to me a similar solution to what I proposed [above](https://github.com/spyder-ide/ux-improvements/issues/75#issuecomment-1233546059), but instead of display a widget, we would display a question mark icon next to an option that would show its associated help on hover.

@conradolandia, could you post a screenshot of what you showed me? Thanks!

**Comment by conradolandia** [comment]
Sure thing. You should see a tool-tip when hovering/clicking the question mark icon. 

![Connection dialog test](https://user-images.githubusercontent.com/5027583/234648562-b9079550-a4f8-41a1-986b-177fb15f3414.png)

**Comment by CAM-Gerlach** [comment]
Yeah, the latter approach follows modern UI patterns and user expectations a lot more closely than the former, and is also a lot more discoverable and easier to use as well, IMO.

**Comment by dalthviz** [comment]
Note: Discussion and definition about the preferences specific organization and text still needs to be done

**Comment by ccordoba12** [comment]
Yep, @CAM-Gerlach is going to take care of that.
